### PR TITLE
Supervisor implementation of Master Proxy

### PR DESCRIPTION
**Pull request by ACBullen**
The stuff I made and merged to my personal fork. Ready for review/use

### CODE REVIEW

File: lib/master_proxy/master_proxy.ex [hunk 1]
```diff
@@ -44,13 +49,28 @@ defmodule MasterProxy do
     Supervisor.start_link(children, supervisor_opts)
   end
 
-  defp load_runtime_config(options) do
-    if Keyword.keyword?(options) do
-      Enum.each(options, fn {key, value} ->
-        Application.put_env(:master_proxy, key, value)
+  defp merge_runtime_and_compiled_opts(options) do
```
Comment (QuinnWilton) [inline lib/master_proxy/master_proxy.ex]: You could rewrite this as:

```elixir
Keyword.merge(Application.get_all_env(:master_proxy), options)
```

**Comment by QuinnWilton** [comment]
I made one comment, but otherwise this looks great to me!

File: lib/master_proxy/master_proxy.ex [hunk 2]
```diff
@@ -49,22 +49,6 @@ defmodule MasterProxy do
...
-        set_opt_or_skip(key, acc, compiled_opts)
-      end)
-
-    relevant_keys
-    |> Enum.reduce(env_configed, fn key, acc ->
-      set_opt_or_skip(key, acc, options)
-    end)
-  end
-
   def set_opt_or_skip(key, target_list, source_list) do
```
Comment (QuinnWilton) [inline lib/master_proxy/master_proxy.ex]: I think this function is unused now.

Also, most idiomatic would probably be moving this file up one directory (lib/master_proxy.ex). What are your thoughts on that?

Comment (ACBullen) [reply lib/master_proxy/master_proxy.ex]: Ah right, and that seems fine to me.

**Pull request merged**